### PR TITLE
[AIRFLOW-4385] Add namespace into pod's yaml

### PR DESCRIPTION
**Pull request by nikolayskovpin**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-4385

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:
Pod's yaml doesn't contain any info regarding namespace

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason: minor changes

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

File: airflow/contrib/kubernetes/kubernetes_request_factory/pod_request_factory.py [hunk 1]
```diff
@@ -75,6 +77,7 @@ class ExtractXcomPodRequestFactory(KubernetesRequestFactory):
     _yaml = """apiVersion: v1
 kind: Pod
 metadata:
+  namespace: default
```
Comment (Erymenn) [inline airflow/contrib/kubernetes/kubernetes_request_factory/pod_request_factory.py]: does that give "default" or "dev" as default namespace?

**Review by Erymenn — APPROVED** [review]

File: airflow/contrib/kubernetes/kubernetes_request_factory/pod_request_factory.py (hunk 1 above)
Comment (nikolayskovpin) [reply airflow/contrib/kubernetes/kubernetes_request_factory/pod_request_factory.py]: default

**Comment by dimberman** [comment]
@nikolayskovpin overall this LGTM. Is there any test you can add against this though? e.g. a dag that broke but would work now?

**Comment by dimberman** [comment]
@nikolayskovpin could you also please rebase?

**Comment by OmerJog** [comment]
@nikolayskovpin there are errors:
```
======================================================================
31) ERROR: Failure: ImportError (cannot import name 'options_pb2')
----------------------------------------------------------------------
   Traceback (most recent call last):
    /usr/local/lib/python3.6/site-packages/nose/failure.py line 39 in runTest
      raise self.exc_val.with_traceback(self.tb)
    /usr/local/lib/python3.6/site-packages/nose/loader.py line 418 in loadTestsFromName
      addr.filename, addr.module)
    /usr/local/lib/python3.6/site-packages/nose/importer.py line 47 in importFromPath
      return self.importFromDir(dir_path, fqname)
    /usr/local/lib/python3.6/site-packages/nose/importer.py line 94 in importFromDir
      mod = load_module(part_fqname, fh, filename, desc)
    /usr/local/lib/python3.6/imp.py line 235 in load_module
      return load_source(name, filename, file)
    /usr/local/lib/python3.6/imp.py line 172 in load_source
      module = _load(spec)
    <frozen importlib._bootstrap> line 684 in _load
      
    <frozen importlib._bootstrap> line 665 in _load_unlocked
      
    <frozen importlib._bootstrap_external> line 678 in exec_module
      
    <frozen importlib._bootstrap> line 219 in _call_with_frames_removed
      
    tests/contrib/operators/test_gcp_spanner_operator.py line 24 in <module>
      from airflow.contrib.operators.gcp_spanner_operator import \
    airflow/contrib/operators/gcp_spanner_operator.py line 24 in <module>
      from airflow.contrib.hooks.gcp_spanner_hook import CloudSpannerHook
    airflow/contrib/hooks/gcp_spanner_hook.py line 25 in <module>
      from google.cloud.spanner_v1.database import Database
    /usr/local/lib/python3.6/site-packages/google/cloud/spanner_v1/__init__.py line 23 in <module>
      from google.cloud.spanner_v1.client import Client
    /usr/local/lib/python3.6/site-packages/google/cloud/spanner_v1/client.py line 31 in <module>
      from google.cloud.spanner_admin_database_v1.gapic.database_admin_client import (  # noqa
    /usr/local/lib/python3.6/site-packages/google/cloud/spanner_admin_database_v1/gapic/database_admin_client.py line 44 in <module>
      from google.iam.v1 import options_pb2
   ImportError: cannot import name 'options_pb2'
```

**Comment by dimberman** [comment]
@OmerJog that doesn't look like a related bug. Is master broken?

**Comment by OmerJog** [comment]
@ashb @potiuk can the tests be restarted?

**Comment by potiuk** [comment]
It think those are real issues. I do not think restart wil help. There were renames of google operators recently and maybe they caused some problems - master does not seem to be broken.

**Comment by nikolayskovpin** [comment]
Yeah, I don't think that my changes caused the tests to fail

**Comment by potiuk** [comment]
You are 243 commits behind current master. So no wonder things might not work @nikolayskovpin :

"This branch is 1 commit ahead, 243 commits behind apache:master.'

Simply rebase to the current master and make sure the currently failed tests are working (maybe they will simply work after rebasing). You can use the recently released BREEZE environment to easily run tests locally. .

**Comment by OmerJog** [comment]
@nikolayskovpin any progress with rebasing?

**Comment by nikolayskovpin** [comment]
> @nikolayskovpin any progress with rebasing?

As I see from master branch, kubernetes module was completely rebuilt. Pod generation is done by kubernetes python library. So I don't know will vertical-pod-autoscaler work with this api properly or not

**Pull request closed without merging**